lib/EnsEMBL/REST/Model/LDFeatureContainer.pm

```diff
@@ -81,6 +81,41 @@ sub fetch_LDFeatureContainer_variation_name {
   return $self->to_array($ldfc);
 }
 
+sub fetch_LDFeatureContainer_slice {
+  my ($self, $slice) = @_;
+  Catalyst::Exception->throw("No region given. Please specify a region to retrieve from this service.") if ! $slice;
+  Catalyst::Exception->throw("Specified region is too large. Maximum allowed size for region is 1Mb.") if ($slice->length > 1_000_000);
+  my $c = $self->context();
+  my $species = $c->stash->{species};
+
+  my $ldfca = $c->model('Registry')->get_adaptor($species, 'Variation', 'LDFeatureContainer');
+
+  my $ld_config = $c->config->{'Model::LDFeatureContainer'};
+  if ($ld_config && $ld_config->{use_vcf}) {
+    $ldfca->db->use_vcf($ld_config->{use_vcf});
+    $Bio::EnsEMBL::Variation::DBSQL::VCFCollectionAdaptor::CONFIG_FILE = $ld_config->{vcf_config};
+    $ENV{ENSEMBL_VARIATION_VCF_ROOT_DIR} = $ld_config->{dir} if (defined $ld_config->{dir});
+  }
+
+  my $population_name = $c->request->param('population_name');
+  if ($population_name) {
+    my $pa = $c->model('Registry')->get_adaptor($species, 'Variation', 'Population');     
+    my $population = $pa->fetch_by_name($population_name);
+    if (!$population) {
+      Catalyst::Exception->throw("Could not fetch population object for population name: $population_name");
+    }
+    my $ldfc;
+    try {
+      $ldfc = $ldfca->fetch_by_Slice($slice, $population);
+    } catch {
+      warn "caught error: $_";
```

Comment:
warn() is bypassing our logging framework. Please use $c->log()->warn($message) or just throw a Catalyst Exception for the controller to deal with. Then we can see that the controller ignores certain kinds of errors in the controller.